### ISSUE TITLE
Sidebar Footer: Consider making the contents permanent, persistent, and everywhere

### ISSUE THREAD

**Issue by jasmussen** [Automattic/wp-calypso#8885]
In this #8578, we visually improved the design of the sidebar footer. It now looks like this for users with just one blog:

![singleblog](https://cloud.githubusercontent.com/assets/1204802/19569291/84f037f2-96f5-11e6-928e-65b3d9b158ec.png)

... and like this for users with multiple blogs:

![multiblog](https://cloud.githubusercontent.com/assets/1204802/19569321/9cef0ce8-96f5-11e6-82fa-642edef7a268.png)

In the My Sites section, multiblog users will see the "Add New Site" button in the site switcher:

![0263a47e-9603-11e6-82fa-8582a1cf3b68](https://cloud.githubusercontent.com/assets/1204802/19569348/af1842ea-96f5-11e6-93d4-2871ea7176e2.gif)

However, we can make another enhancement, and there's a topic to discuss. First off, the sidebar footer is not yet on the Me section. When the sidebar footer was introduced, and it verbosely spelled out "Help", there was visual redundancy on the Me page. But I feel like with the icon only help icon, this is no longer the case. **Should the sidebar footer be introduced on the Me page now?** 

Second question is regarding consistency. The original goal of #8578 was to make the sidebar footer the same for both single users and multi-users, with no need to enter the switcher to see the "Add New Site" button. 

There were arguments against that <a href="https://github.com/Automattic/wp-calypso/pull/8578#issuecomment-254786421">here</a> and <a href="https://github.com/Automattic/wp-calypso/pull/8578/files/72dec60eb2e9e8ba85513e41ae2055c87c6a4956#r84045941">here</a>.

Aside from being my initial goal, @rickybanister also <a href="https://github.com/Automattic/wp-calypso/pull/8578#issuecomment-255145992">suggested</a> that the sidebar should be the same everywhere, on all sections. The primary benefit of this, as I see it, is that good navigation doesn't change, and by virtue of being a _footer_ more than a _sidebar_, it feels like the left side of the footer shouldn't be contextual. 

We should try and reach a concensus on what to do here.

**Comment by mtias** [comment]
I think the sidebar footer should be permanent within a given section, the same way the sidebar remains but its content change from section to section. I don't think it makes to make it the same across the board because then it competes with the masterbar in terms of IA and gives things like "add new site" too much prominence in places where ti doesn't make sense (like the editor).

My proposal for the sidebar footer:

- Help and Chat across all sections.
- My Sites, single and multi, could show "add new site".
- Editor, could have "add new post" in current site (incidentally, a missing flow).
- Reader, could have "follow new site" (open to other ideas).
- Me, is tricky, I propose we show footer and remove sidebar item for help. We should highlight the help button as "selected" in the footer when you are in `/help` if we do this.
- Then we need to see what could be the primary footer action. Perhaps "log out", but not sure.

**Comment by rickybanister** [comment]
I think it will feel weird/confusing if help/chat were omnipresent, but the button next to them changes slightly from section to section. Especially if people already click 'add new site' expecting to add a page to their site (an anecdote I heard from @meremagee). 

If we only want to have 'add site' available within 'my sites' routes, that sounds logical, but perhaps that's enough of a first-class citizen that we just leave that spot blank in the other sections.

**Comment by jasmussen** [comment]
The need for a consistent and permanent sidebar footer came up in an only tangentially related issue, which I've worked on in #8918. That PR shows the sidebar on the Me section, and as the screenshot shows, I think it looks good.

Unless anyone on this thread objects to that being merged in, then it seems like half of this issue will be solved.

**Comment by mtias** [comment]
> If we only want to have 'add site' available within 'my sites' routes, that sounds logical, but perhaps that's enough of a first-class citizen that we just leave that spot blank in the other sections.

I do want to explore having the "add new post" button in the editor in that place — it's a flow that I've realized I miss fairly often and now there is a place to put it that doesn't seem awkward.

**Comment by folletto** [comment]
>  But I feel like with the icon only help icon, this is no longer the case. Should the sidebar footer be introduced on the Me page now?

I think the sidebar footer, as a component, should be shown everywhere the sidebar is present.

> I think the sidebar footer should be permanent within a given section, the same way the sidebar remains but its content change from section to section.

I find this model good for our scoping here.

> Then we need to see what could be the primary footer action. 

The only thing here for me is that we shouldn't force elements inside the empty slot in the footer just because we have it. It's ok to have it empty, and once the need arises, a real need, we know we can make use of it.

I feel also that having sections where it's empty will make the sections where it's present even more clear as the change would be more visible. So I agre with @rickybanister in just adding the item as "Add Site", and leave it empty for now — until a strong need arises.

**Comment by rickybanister** [comment]
I agree with @folletto's agreeing with me. I'd rather reinforce the masterbar's add post action in some way. 

Could we gather data on something like: a user is viewing the editor, they click 'add post' from the masterbar, and detect how often they add an additional post to the same site vs a different site? 

Currently when you click the 'add post' button your current site is highlighted, but it is mixed into the list. Could we try putting the current site at the top of the list—that might not remove a click, but require less scanning and mouse/finger moving.

Basically I think we could solve this 'problem' in a simpler way with tools we already have.

**Comment by mtias** [comment]
> Basically I think we could solve this 'problem' in a simpler way with tools we already have.

My argument is that since we have a footer where we are placing an "add new item related to this domain" the simplest way would be to use it. I don't think "New Post" in the masterbar can elegantly support all these context changes.

**Comment by mtias** [comment]
> Could we gather data on something like: a user is viewing the editor, they click 'add post' from the masterbar, and detect how often they add an additional post to the same site vs a different site?

Do we _need_ to gather this? As in, what would it hurt to support it? Even if it's just a few users, I don't see why we wouldn't want to support this flow somehow if we have the tools to do so. (I believe @ryanboren has brought it up before as impacting his posting behaviour.)
